classes/script_metadata.php

```diff
@@ -209,18 +209,38 @@ public static function get_redactable_param_names(): array {
      *
      * @return string the request path for this profile.
      */
-    public static function get_request(): string {
-        global $SCRIPT, $ME, $CFG;
+    public static function get_normalised_relative_script_path(): string {
+        global $SCRIPT;
 
-        if (!isset($ME)) {
-            // If set, it will trim off the leading '/' to normalise web & cli requests.
-            $request = isset($SCRIPT) ? ltrim($SCRIPT, '/') : self::REQUEST_UNKNOWN;
-            return $request;
+        // This is from the previous iteration of the function.
+        if (!isset($SCRIPT)) {
+            return self::REQUEST_UNKNOWN;
         }
 
-        $request = (new \moodle_url($ME))->out_omit_querystring();
-        $request = str_replace($CFG->wwwroot, '', $request);
+        $request = $SCRIPT;
+        $count = 0;
+
+        // Remove multiple consecutive slashes.
+        do {
+            $request = str_replace('//', '/', $request, $count);
```

Comment:
maybe some personal preference here but I reckon this might be simpler as a preg_replace without needing a loop

Reply:
I thought about it and I agree.